driver-core/src/main/com/mongodb/internal/logging/LogMessage.java

```diff
@@ -41,11 +44,34 @@ public final class LogMessage {
     private final String format;
 
     public enum Component {
-        COMMAND,
-        CONNECTION
+        COMMAND("command"),
+        CONNECTION("connection"),
+        SERVER_SELECTION("serverSelection");
+
+        private static final Map<String, Component> INDEX;
+
+        static {
+            INDEX = Stream.of(Component.values()).collect(Collectors.toMap(Component::getValue, identity()));
+        }
+
+        private final String value;
+
+        Component(final String value) {
+            this.value = value;
+        }
+
+        public String getValue() {
+            return value;
+        }
+
+        public static Component of(final String value) {
+            Component result = INDEX.get(value);
+            return assertNotNull(result);
+        }
```

Comment:
The `serverSelection` component is a bit more complex than the previous one-word ones, so more code is needed to match components and decode a string to `Component`.

Reply:
It seems the only usage of this INDEX map is in tests. We could consider including it as part of test fixtures, perhaps within the Entities.class to keep it separate for code clarity and the main logic a bit less cluttered.

Reply:
I marked the `getValue`, `of` methods with `@VisibleForTesting(otherwise = PRIVATE)` (I should have done this from the get go). If `Component` were API, it would have indeed been reasonable not to add these methods to `Component`. However, given that the code is internal, it seems to me that having this methods on `Component` itself is better, because conceptually they seem to belong on it.